test/unit/LoopbackModelBase.test.js

```diff
@@ -0,0 +1,48 @@
+const { expect } = require('chai');
+
+const LoopbackModelBase = require('../../src/LoopbackModelBase');
+
+const MockApp = require('../support/MockApp');
+
+describe('The LoopbackModelBase Class', function() {
+    it('throws an error if no valid loopback model is given', function(){
+        try {
+            new LoopbackModelBase({ });
+        } catch (error) {
+            expect(error).to.have.property('message', 'No Loopback Model Instance was given to the\n                LoopbackModelBase constructor');
+        }
+    });
+
+    it('creates an instance with the given values', function(){
+        const instance = new LoopbackModelBase({ model: {}, modelName: 'test' });
+
+        expect(instance).to.have.property('modelName', 'test');
+    });
+
+    it('throws an error when calling get env without loopback booted', function(){
+        const instance = new LoopbackModelBase({ model: {}, modelName: 'test' });
+
+        try {
+            instance.getEnv();
+        } catch (error) {
+            expect(error).to.have.property('message', '\n                    The loopback application environment could\n                    not be loaded because the application was not initalized yet.\n                    Therefore not app object is avalialble.\n                    Usualy this means you need to call the looback boot function');
+        }
+    });
+
+    it('createError returns an error ', function(){
+        const error = LoopbackModelBase.createError('test error');
```

Comment:
Just a hint: having a method to create errors will always add an additional entry onto your stack trace which can be confusing.